test/extensions/transport_sockets/tls/integration/ssl_integration_test.cc

```diff
@@ -344,48 +344,69 @@ class SslTapIntegrationTest : public SslIntegrationTest {
   void initialize() override {
     // TODO(mattklein123): Merge/use the code in ConfigHelper::setTapTransportSocket().
     config_helper_.addConfigModifier([this](envoy::config::bootstrap::v2::Bootstrap& bootstrap) {
-      auto* filter_chain =
-          bootstrap.mutable_static_resources()->mutable_listeners(0)->mutable_filter_chains(0);
-      // Configure inner SSL transport socket based on existing config.
-      envoy::api::v2::core::TransportSocket ssl_transport_socket;
-      ssl_transport_socket.set_name("tls");
-      MessageUtil::jsonConvert(filter_chain->tls_context(), *ssl_transport_socket.mutable_config());
-      // Configure outer tap transport socket.
-      auto* transport_socket = filter_chain->mutable_transport_socket();
-      transport_socket->set_name("envoy.transport_sockets.tap");
-      envoy::config::transport_socket::tap::v2alpha::Tap tap_config;
-      tap_config.mutable_common_config()
-          ->mutable_static_config()
-          ->mutable_match_config()
-          ->set_any_match(true);
-      auto* output_config =
-          tap_config.mutable_common_config()->mutable_static_config()->mutable_output_config();
-      if (max_rx_bytes_.has_value()) {
-        output_config->mutable_max_buffered_rx_bytes()->set_value(max_rx_bytes_.value());
+      if (upstream_tap_) {
```

Comment:
Sorry -- meant to delete this comment, and replace it with this one:

WDYT of splitting out a separate initializeUpstreamTap() method which is called just from the new test? Might be easier to read.

Reply:
Unfortunately it's not that easy because the integration test initialize function gets called from a different embedded function. I can add some more helper routines which will make it easier to read.